File: solutions/2571-find-the-pivot-integer/solution.py
class Solution(object):
    def pivotInteger(self, n):
        """
        :type n: int
        :rtype: int
        """
        left=n*(n+1)/2
        right=0
        for i in range(1,n+1):
            left-=i
            if left==right:
                return i
            right+=i
        return -1

File: solutions/2571-find-the-pivot-integer/solution.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int a=((n+1)*n)/2;
        int b=0;
        for(int i=1;i<=n;i++)
        {
            a-=i;
            if(a==b)
                return i;
            b+=i;
        }
        return -1;
    }
};
